server/controllers/chartsController/helpers.js

```diff
@@ -303,3 +331,67 @@ export const calculateSubjectVariableDayCount = (
 
   return subjectsDayDataAndVariable.length
 }
+
+export const buildFacetForFilters = ({
+  isSexAtBirthFilterActive,
+  isInclusionCriteriaFilterActive,
+  userAccess,
+}) => {
+  const facetForFilters = {}
+
+  if (isSexAtBirthFilterActive) {
+    facetForFilters.socioDemographics = [
+      {
+        $match: {
+          assessment: SOCIODEMOGRAPHICS_FORM,
+          study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+        },
+      },
+      {
+        $project: {
+          ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+          socioDemographicsCollection: MONGO_COLLECTION_STRING,
+        },
+      },
+    ]
+  }
+  if (isInclusionCriteriaFilterActive) {
+    facetForFilters.inclusionCriteria = [
+      {
+        $match: {
+          assessment: INCLUSION_EXCLUSION_CRITERIA_FORM,
+          study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+        },
+      },
+      {
+        $project: {
+          ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+          inclusionCriteriaCollection: MONGO_COLLECTION_STRING,
+        },
+      },
+    ]
+  }
+
+  return facetForFilters
+}
+
+export const intersectSubjectsFromFilters = (
+  filters,
+  mapForResultsFromFilters
+) => {
+  const [filterListA, filterListB] = Object.values(filters).map((filter) => {
```

Comment:
What is the shape of the `filters` object? It seems like the value has subject data, what are the `keys`?

Reply:
The requested filters is the result from the filters facet 
```suggestion
  {
        socioDemographics: [
          {
            study: 'YA',
            subject: 'YA01508',
            socioDemographicsCollection:
              '99fc2c55c6718fcd8eadf9244a24d2f68aae77c6fc591d5e2a8e67bab906446f',
          },
        ],
        inclusionCriteria: [
          {
            study: 'YA',
            subject: 'YA01508',
            inclusionCriteriaCollection:
              '062419c48a8b6634e696b9dc0674fb87c34aea44751d1eba107feb0fc7325d71',
          },
          {
            study: 'CA',
            subject: 'CA00063',
            inclusionCriteriaCollection: 'CA00063filter',
          },
          {
            study: 'LA',
            subject: 'LA00028',
            inclusionCriteriaCollection: 'la00028filter',
          },
          {
            study: 'YA',
            subject: 'YA00015',
            inclusionCriteriaCollection: 'YA00015Filter',
          },
        ],
      }
```

Reply:
```
 { 
   filterA: [
     {
        subject,
        filterACollection,
        study
    }
  ],
  filterB: [
      {
        subject,
        filterBCollection,
        study
      }
  ]
```

Reply:
And with Object.values() the object changes into an array of arrays